packages/loader/container-loader/src/container.ts

```diff
@@ -935,6 +935,78 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
     }
 
+    public connect() {
+        if (!this.closed && !this.connected) {
+            // Note: no need to fetch ops as we do it preemptively as part of DeltaManager.attachOpHandler().
+            // If there is gap, we will learn about it once connected, but the gap should be small (if any),
+            // assuming that connect() is called quickly after initial container boot.
+            this.connectInternal({ reason: "DocumentOpenResume", fetchOpsFromStorage: false });
+        }
+    }
+
+    private connectInternal(args: IConnectionArgs) {
+        assert(!this.closed, "Attempting to connect() a closed DeltaManager");
```

Comment:
I'd like to see code reuse with existing resume/setAutoReconnect flow to easier see where this logic is different from existing logic. Ideally there should be no differences and one implementation should call into another, right?

Reply:
So basically have `connect()` call `resumeInteranl()` and `setAutoReconnect(true)`? If so, we should probably make `setAutoReconnect()` a private function (since we would no longer be removing it).

Reply:
It does not matter which way you go (connect calling setAutoReconnect(true) or other way around). We can clean that later when we remove deprecated APIs (and have just one function), but for now it would be really great to see if new functionality differs from the previous one through code reuse. If you can't reuse code, that means it's not 1-to-1 reman, and it should be clearly spelled out for consumers such that they are not caught by suprise.

Reply:
The reason why I wrote a new `connectInernal()` function is because I felt that what we wanted is an overlap between the current `resuem()` and `setAutoReconnect(true)` functions. However, I didn't want to call those functions straight up because there is a bit of overlap between the two. For example, they both call end up calling `this.connectToDeltaStream()`. Although I don't think this would cause issues it did seem a bit messy so I ended up essentially combining the functions while trying to eliminate the overlap (which resulted in `connectInternal()`). So I do believe we could reuse code, but it would be a bit inefficient. Do you think this is a bad approach and we should just reuse the existing logic?

Reply:
So I assume the guidance is for people to move from setAutoReconnect(true) to connect() calls.
The two obvious things I see that are different in flows:
1. There is a check for _attachState === AttachState.Attached && this.resumedOpProcessingAfterLoad in setAutoReconnect() that is missing here
2. There is manualReconnectInProgress tracking.

Second one is likely not interesting and can probably be removed (or added).
First part - I do not know, it feels like it should apply here (I'd look at history on when/why it was added)

Reply:
> 1. There is a check for _attachState === AttachState.Attached && this.resumedOpProcessingAfterLoad in setAutoReconnect() that is missing here
> 2. There is manualReconnectInProgress tracking.

Thank you for catching those. It looks like there may be issues calling `this.connectToDeltaStream()` if the container is not attached. We can add an assertion to the start of `connectInternal()` to ensure that the container is attached.

As per `resumedOpProcessingAfterLoad`, we won't need to check this because we are already ensuring that the container is resumed as part of the `connectInternal()` flow.

`manualReconnectInProgress` can also be added, we just need to make sure it is being set when it is manual connection and not part of the loading/attatch() flow.

Reply:
Looks reasonable after latest changes.

Reply:
I think you could call `resumeInternal` from the top of `connectInternal` now, would be nice to reduce code duplication (and `resumeInternal` looks like it will stick around since it's called from `attach`)

Reply:
And totally optional, but consider pulling the `if (currentMode !== mode) { ... } ` block into a helper, since it's now here in triplicate.  And it's pretty crucial the three places stay in sync since they're messing with the same state.

Reply:
+1 to putting `manualReconnectInProgress` back the way it was... Or if not, then just delete that variable because this was the only place it was set to true.

Probably should take a pass over the values used for `connectionInitiationReason` in `logConnectionStateChangeTelemetry` and see if they still make sense given the merging of `resume` and `setAutoReconnect(true)` from an API standpoint.  I think I'm confused why "AutoReconnect" and "ManualReconnect" were distinguished by this case anyway...

Reply:
I agree with your first two comments and will update the PR shortly to make these changes. Re: your last comment, since we merged `resume()` and `setAutoReconnect()`, I think it does make sense to just delete `manualReconnectInProgress`.